### PR TITLE
Add image build and push make targets

### PR DESCRIPTION
**Pull request by mikenairn**
Changes required to install the operator as part of integreatly:

* Additional make targets to build and push image to quay.io
* Updated roles/rbacs for cluster level permissions
* Update Dockerfile

Related:

* https://github.com/integr8ly/installation/pull/1161

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+COPY build/_output/bin/heimdall ${OPERATOR}
 
-#USER nobody
+COPY build/bin /usr/local/bin
+RUN  /usr/local/bin/user_setup
 
-#ADD build/_output/bin/heimdall /usr/local/bin/heimdall
+ENTRYPOINT ["/usr/local/bin/entrypoint"]
+
+USER ${USER_UID}
```
Comment (mikenairn) [inline build/Dockerfile:15]: Used the same base image and setup that we have used in other operators.

File: deploy/crds/imagemonitor_v1alpha1_imagemonitor_cr.yaml [hunk 2]
```diff
@@ -3,5 +3,4 @@ kind: ImageMonitor
 metadata:
   name: example-imagemonitor
 spec:
-  # Add fields here
-  namespaces: 3
+  excludePattern: ""
```
Comment (mikenairn) [inline deploy/crds/imagemonitor_v1alpha1_imagemonitor_cr.yaml:6]: This actually stops anything working, currently it looks like you need to set an excludePattern to something or it excludes everything. .i.e `excludePattern: "thismatchesnothing"`. Removing it entirely also results in an empty one being added, which has the same result.

File: deploy/operator.yaml [hunk 3]
```diff
@@ -15,8 +15,7 @@ spec:
       serviceAccountName: heimdall
       containers:
         - name: heimdall
-          # Replace this with the built image name
-          image: REPLACE_IMAGE
+          image: quay.io/integreatly/heimdall-operator:master
```
Comment (mikenairn) [inline deploy/operator.yaml:18]: Went with the same pattern as other operators here where the image in this file is set to the latest released version. Will keep it as master until we are ready to do an initial release.

File: deploy/operator.yaml [hunk 4]
```diff
@@ -33,9 +32,7 @@ spec:
             failureThreshold: 1
           env:
             - name: WATCH_NAMESPACE
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.namespace
+              value: ""
```
Comment (mikenairn) [inline deploy/operator.yaml:35]: watch all namespaces

File: deploy/crds/imagemonitor_v1alpha1_imagemonitor_cr.yaml (hunk 2 above)
Comment (maleck13) [reply deploy/crds/imagemonitor_v1alpha1_imagemonitor_cr.yaml:6]: This is fixed in the other PR

**Review by maleck13 — APPROVED** [review]
looks good to me

**Pull request merged**